### PR TITLE
hard crash if one Xpress function not found

### PR DESCRIPTION
**Pull request by sgatto**
ortools by google uses a dynamic_library that performs a hard check on the successful load of every function.
RTE developed a different, softer approach that is more refined.
Here I propose to revert to the google version and drop RTE version in order for our development to be more easily accepted in the official repository.

### CODE REVIEW

**Comment by sgatto** [comment]
The latest development by google make the Python 3.8 CI actions to crash.

File: ortools/xpress/environment.cc [hunk 1]
```diff
@@ -244,7 +236,9 @@ absl::Status LoadXpressDynamicLibrary(std::string& xpresspath) {
     }
 
     if (xpress_library.LibraryIsLoaded()) {
-      xpress_load_status = LoadXpressFunctions(&xpress_library);
+      LOG(INFO) << "Loading all Xpress functions";
+      LoadXpressFunctions(&xpress_library);
```
Comment (flomnes) [inline ortools/xpress/environment.cc:240]: Error handling isn't so great here. Shouldn't we return an `absl::status` here ?

**Comment by sgatto** [comment]
> Suggestion: If one function is missing, mark XPRESS as unavailable instead of a "hard crash".

I have tried it, but I can't figure out how to catch the exception thrown by the `CHECK` macro [here](https://github.com/rte-france/or-tools/blob/bb41ef405f85175d1658cadd858796a02c0ea540/ortools/base/dynamic_library.h#L70C8-L70C8)

I reproduced the same pattern used for the Gurobi interface

File: ortools/xpress/environment.cc (hunk 1 above)
Comment (sgatto) [reply ortools/xpress/environment.cc:240]: I cannot find a way to do it.
`LoadXpressFunctions` calls `DynamicLibrary.GetFunction(const char* function_name) `

`DynamicLibrary.GetFunction(const char* function_name) ` executes the `CHECK` macro that seems to throw a fatal error. I tried to catch it but i probably did not figure out how to do it properly.
Gurobi interface uses the same approach you see here and in the end we can consider that
not having Xpress installed is ok
having the wrong xpress installation is worse

Comment (sgatto) [reply ortools/xpress/environment.cc:240]: I open to change this behavior but at the moment I cannot see how to do it without touching the `dynamic_library.h` file

**Review by pet-mit — APPROVED** [review]

**Review by flomnes — APPROVED** [review]

**Pull request merged**